lib/publiccloud/provider.pm

```diff
@@ -337,8 +423,22 @@ Destroys the current terraform deployment
 sub terraform_destroy {
     my ($self) = @_;
     record_info('INFO', 'Removing terraform plan...');
-    assert_script_run('cd ' . TERRAFORM_DIR);
-    script_run('terraform destroy -no-color -auto-approve', TERRAFORM_TIMEOUT);
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
+    }
+    else {
+        assert_script_run('cd ' . TERRAFORM_DIR);
+    }
+    my $ret = script_run('terraform destroy -no-color -auto-approve', TERRAFORM_TIMEOUT);
+    unless (defined $ret) {
+        type_string(qq(\c\\));        # Send QUIT signal
+        assert_script_run('true');    # make sure we have a prompt
+        record_info('ERROR', 'Terraform destroy failed with timeout', result => 'fail');
+        assert_script_run('cd ' . TERRAFORM_DIR);
+        $self->on_terraform_destroy_timeout();
+        die('Terraform destroy failed with timeout');
```

Comment:
I think we didn't died for reason. As we are in cleanup phase and this would avoid revoking the credentials from vault.
If you wanna track this state, lets set result to 'failed'. WDYT?

Reply:
So replace the two `die` commands in this function by something like `record_info('ERROR', 'Terraform destroy failed with timeout', result => 'fail');`?